output/datadog/messagepacker.go

```diff
@@ -0,0 +1,113 @@
+package datadog
+
+import (
+	"bytes"
+
+	"github.com/klauspost/compress/gzip"
+	"github.com/relex/gotils/logger"
+	"github.com/relex/slog-agent/base"
+	"github.com/relex/slog-agent/defs"
+	"github.com/relex/slog-agent/util"
+)
+
+const gzipCompressionLevel = gzip.BestSpeed
+
+// outputChunkMaxDataBytes defines the max uncompressed data size of a LogChunk
+// The value must be lower than the maximum buffer length acceptable by upstream
+var outputChunkMaxDataBytes = 5 * 1024 * 1024
+
+// outputChunkMaxRecords defined the maximum amount of log entries in a single chunk for datadog output
+var outputChunkMaxRecords = 1000
```

Comment:
A link to the documented limits may be added here: https://docs.datadoghq.com/api/latest/logs/#send-logs